core/src/com/tgf/twf/core/world/storage/Storage.java

```diff
@@ -53,18 +43,68 @@ public int getRemainingCapacity(final ResourceType resourceType) {
     }
 
     /**
-     * @param resources The quantity of resources to consume, per resource type.
-     * @return true if the storage had enough resources in store to satisfy the demand. If true, the resources are removed from the storage. If
-     * false, the storage is left untouched.
+     * @param resources The quantity of resources to store, per resource type.
+     * @return true if the storage have enough capacity in store to satisfy the offer.
      */
-    public boolean tryConsumeResources(final Inventory resources) {
-        final boolean canConsume = resources.getStoredResourceTypes().stream()
-                .allMatch(resourceType -> inventory.getStoredQuantity(resourceType) >= resources.getStoredQuantity(resourceType));
-        if (canConsume) {
-            resources.getStoredResourceTypes()
-                    .forEach(resourceType -> inventory.retrieve(resourceType, resources.getStoredQuantity(resourceType)));
+    public boolean canStore(final Inventory resources) {
+        return resources.getStoredResourceTypes().stream()
+                .allMatch(resourceType -> capacity.getRemainingCapacity(inventory, resourceType) >= resources.getStoredQuantity(resourceType));
+    }
+
+    /**
+     * @param resources The quantity of resources to store. {@link #canStore(Inventory)} (Inventory)} should be called first to check if the
+     *                  storage can accept the offer or not. {@link Capacity} overflow will throw an {@link IllegalStateException}.
+     */
+    public void store(final Inventory resources) {
+        if (!inventory.store(resources)) {
+            throw new IllegalStateException("Storage rejected storage of " + resources);
+        }
+    }
+
+    /**
+     * @param resources The quantity of resources to reservation. {@link #canStore(Inventory)} (Inventory)} should be called first to check if the
+     *                  storage can accept the offer or not. {@link Capacity} overflow will throw an {@link IllegalStateException}.
+     *                  The effect of this call is to reserve capacity in the storage for a future {@link #store(Inventory)} call.
```

Comment:
Nitpick: maybe everything from the second sentence onwards can be in the summary instead of the @param tag? You could also add a @throws tag.

Reply:
Ok